### ISSUE TITLE
Routes scaffolding fails when running Super Scaffolding test serially.

### ISSUE THREAD

**Issue by andrewculver** [bullet-train-co/bullet_train#192]
@gazayas I wonder if this is a bug you've already fixed on another repo but we didn't merge in before The Great Unbundling.

**Comment by gazayas** [comment]
@andrewculver Looking over the code, it looks like this is the same problem we were facing that I addressed in [PR #591](https://github.com/bullet-train-co/bullet-train-tailwind-css/pull/591) in the other repository. I'll start working with that to implement it here. Before that, I just wanted to get some clarity on the routes in the Super Scaffolding test.

For these models, I'm under the impression that we need these routes:
```shell
spring rails g model Insight team:references name:string
spring rails g model Personality::CharacterTrait insight:references name:string

spring rails g model Personality::Disposition team:references name:string
spring rails g model Personality::Note disposition:references name:string
spring rails g model Personality::Observation team:references name:string
spring rails g model Personality::Reactions::Response observation:references name:string
```

```ruby
resources :teams do
  resources :insights do
    namespace :personality do
      resources :character_traits
    end
  end

  namespace :personality do
    resources :dispositions do
      resources :notes
    end
    resources :observations do
      namespace :reactions do
        resources :responses
      end
    end
  end
end
```

So, for example, we WOULDN'T write the following routes for `Personality::Note` because it would be under the same namespace as `Personality::Disposition`. Both `Disposition` and `Note` refer to the same namespace, so we wouldn't make an entire new namespace like this:
```ruby
resources :teams do
  namespace :personality do
    resources :dispositions do
      namespace :personality do
        resources :notes
      end
    end
  end
end
```

This is determined mainly by the model being referenced (i.e. - `Personality::Disposition` under `Team`, and `Personality::CharacterTrait` under `Insight`).

I just wanted to check with you if all of that makes sense. I'll start working on a fix 🛠️

**Comment by gazayas** [comment]
@andrewculver Working specifically with [PR #31](https://github.com/bullet-train-co/bullet_train-super_scaffolding/pull/31) in `bullet_train-super_scaffolding`, tests aren't working serially because of the different namespaces/resources we're trying to add all at once.

With the parallel tests, since the nodes split the scaffolding process, `config/routes.rb` doesn't get convoluted with different blocks so everything works fine. If you the run commands in the `if` blocks in the setup script one at a time and run the Super Scaffolding system test, everything passes locally just fine. It's just when we try to add everything all at once which is when the trouble starts.

I actually really like the setup which only scaffolds the models for a specific node, but we unfortunately might miss some bugs in the `RoutesFileManipulator` if we don't run things serially and put all of our routes in `config/routes.rb`.

The system tests work fine with the fix in the PR above, but minitest is throwing an error so I'll keep working on that.
